query-service-impl/src/main/java/org/hypertrace/core/query/service/RequestHandler.java

```diff
@@ -27,4 +27,9 @@ public interface RequestHandler {
 
   /** Handle the request and add rows to the collector. */
   Observable<Row> handleRequest(QueryRequest request, ExecutionContext executionContext);
+
+  default QueryRequest applyAdditionalFilters(
```

Comment:
There is already an existing step of query transformation. Look at `QueryTransformation`. This step of adding filters should go there.

Reply:
We started from this, then realised we don't get scope information in query request.
If we apply a filter on Transformation step, it'll be applied on all the scopes and would work as global filter. Which is wrong behaviour
Query Service decides the scope and it's handler  based of the query. Hence pushed down these filters to handler